spec/factories.rb

```diff
@@ -410,7 +422,18 @@
   end
 
   factory :line_item_with_shipment, parent: :line_item do
-    target_shipment { create(:shipment, order: order) }
+    transient do
+      shipping_fee 3
+    end
+
+    target_shipment do
+      shipment = order.reload.shipments.first
```

Comment:
is there any particular reason why we need a `reload` here? this will run an extra DB query so it'll add an overhead anywhere we use this factory.

Reply:
yes! these factories are all DB dependent :-( and this reload is really required here...